implementations/rust/ockam/ockam_vault_core/src/open_close_vault.rs

```diff
@@ -2,6 +2,12 @@ use crate::vault::Vault;
 use ockam_error::OckamResult;
 
 pub trait OpenCloseVault: Sized {
+    type InnerVault;
+
+    fn get_data_mut(&mut self) -> &mut Self::InnerVault;
```

Comment:
I'm not understanding the purpose for these functions.

Reply:
@mikelodder7 it allows to use vaults like this:
```rs
let mut vault = SoftwareVault::default();
let mut vault = vault.open().unwrap();
let mut attributes = SecretAttributes {
    stype: SecretType::Curve25519,
    persistence: SecretPersistence::Ephemeral,
    length: CURVE25519_SECRET_LENGTH,
};

let res = vault.secret_generate(attributes);
```

So, SoftwareVault implements OpenCloseVault trait, but none of other vault traits. InnerVault is SoftwareVaultImpl and implements all the other vault traits (like sign, verify, encrypt, etc.)